### PR TITLE
[8.0.x] example: Add Monoid

### PR DESCRIPTION
**Pull request by NeQuissimus**
Depends on #1546

### CODE REVIEW

File: example/src/main/tut/typeclass/Monoid.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Monoid
+
+**Category Theory**
+
+*A monoid is a semigroup with a unique identity element.*
+
+A monoid instance must satisfy the following laws in addition to those defined by [Semigroup](Semigroup.html):
+
+- `append(empty, x) === x`
+- `append(x, empty) === x`
```
Comment (hrhino) [inline example/src/main/tut/typeclass/Monoid.md]: Maybe label them left/right identity?

File: example/src/main/tut/typeclass/Monoid.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import scalaz.typeclass.Semigroup._
+
+def ListSemigroup[A]: Semigroup[List[A]] = new Semigroup[List[A]] {
+  def append(a1: List[A], a2: => List[A]) = a1 ++ a2
+}
+
+implicit def ListMonoid[A]: Monoid[List[A]] = new Monoid[List[A]] {
+  val semigroup = ListSemigroup[A]
+  val empty = List.empty[A]
+}
```
Comment (hrhino) [inline example/src/main/tut/typeclass/Monoid.md]: Mentioning `MonoidClass` (which lets you define `append` and `empty` instead) might be useful as well.

File: example/src/main/tut/typeclass/Monoid.md (hunk 1 above)
Comment (NeQuissimus) [reply example/src/main/tut/typeclass/Monoid.md]: Ah, I like that!

File: example/src/main/tut/typeclass/Monoid.md (hunk 2 above)
Comment (NeQuissimus) [reply example/src/main/tut/typeclass/Monoid.md]: I added an additional example

**Comment by jdegoes** [comment]
👍

**Pull request merged**